### PR TITLE
Various improvments to improve local simulation

### PR DESCRIPTION
**Pull request by Vuwij**

### CODE REVIEW

**Review by manx52 — APPROVED** [review]
Seems good

File: soccer_common/src/soccer_common/test_common.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            q = Transformation.get_quaternion_from_euler([cam_angle, 0, 0])
+
+            for cam_position in [[0, 0, 0], [0, 0, 0.1], [0, 0, -0.1]]:
+                p = Transformation(cam_position, q)
+                c = Camera("robot1")
+                c.pose = p
+                c.resolution_x = 360
+                c.resolution_y = 240
+
+                positions = [[0.5, 0, 0.1], [0.5, 0, 0], [0.5, 0, 0.1]]
```
Comment (Shah-Rajab) [inline soccer_common/src/soccer_common/test_common.py:23]: change name to ball_positions.

File: soccer_strategy/src/robot_controlled_3d.py [hunk 2]
```diff
@@ -72,6 +73,18 @@ def set_navigation_position(self, goal_position):
         self.goal_publisher.publish(p)
         return True
 
+    def reset_robot_callback(self, pose: Pose):
```
Comment (Shah-Rajab) [inline soccer_strategy/src/robot_controlled_3d.py:76]: Add comment: ONLY USE FOR TESTING

File: soccer_strategy/src/robot_controlled_3d.py [hunk 3]
```diff
@@ -29,6 +29,7 @@ def __init__(self, team, role, status):
         self.imu_subsciber = rospy.Subscriber("imu_filtered", Imu, self.imu_callback)
         self.action_completed_subscriber = rospy.Subscriber("action_complete", Empty, self.action_completed_callback, queue_size=1)
         self.head_centered_on_ball_subscriber = rospy.Subscriber("head_centered_on_ball", Empty, self.head_centered_on_ball_callback, queue_size=1)
+        self.reset_robot_subscriber = rospy.Subscriber("reset_robot", Pose, self.reset_robot_callback, queue_size=1)
```
Comment (Shah-Rajab) [inline soccer_strategy/src/robot_controlled_3d.py:32]: This signal is asserted during the tests and not during a match.

Comment (Shah-Rajab) [inline soccer_strategy/src/robot_controlled_3d.py:32]: This signal is asserted during the tests and not during a match.

**Pull request merged**